brave/src/main/java/com/linecorp/armeria/client/brave/ArmeriaHttpClientAdapter.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogAvailability;
+import com.linecorp.armeria.internal.brave.SpanTags;
+
+import brave.http.HttpClientAdapter;
+
+public class ArmeriaHttpClientAdapter extends HttpClientAdapter<RequestLog, RequestLog> {
+    /**
+     * Original implementation is calling {@link HttpClientAdapter#url(Object)} which needs
+     * {@link RequestLog#scheme()} is not available at {@link RequestLogAvailability#REQUEST_START}.
+     * We need to use {@link RequestLog#path()} directly.
+     *
+     * @see brave.http.HttpAdapter#path(Object)
+     */
+    @Override
+    public String path(RequestLog requestLog) {
+        return requestLog.path();
+    }
+
+    @Override
+    public String method(RequestLog requestLog) {
+        return requestLog.method().name();
+    }
+
+    @Override
+    public String url(RequestLog requestLog) {
+        return SpanTags.generateUrl(requestLog);
```

Comment:
better to check SCHEME availability?

Reply:
Both `SCHEME` and `REQUEST_HEADERS`. What's the expected behavior of this method when it's called too early (Just curious)?

Reply:
all methods return null if not available (in case of the special cased integer responseCode zero)